bundle/src/main/java/com/adobe/acs/commons/workflow/process/impl/WorkflowDelegationStep.java

```diff
@@ -78,125 +103,217 @@
+
+      this.doDelegate(delegateWorkflowModel, workflowSession, workItem, terminateOnDelegation);
+    }
 
-        if (StringUtils.isBlank(propertyName)) {
-            throw new WorkflowException("PROCESS_ARG [ " + WORKFLOW_PROPERTY_NAME + " ] not defined");
+    /**
+     * Does fallback from 1. conf property 2. content property 3. default property
+     * 
+     * @throws WorkflowException
+     *           if found model id can not be located
+     */
+    private WorkflowModel getDelegateWorkflowModel(WorkflowSession workflowSession, String confWorkflowModelId,
+        String contentWorkflowModelId, String defaultWorkflowModelId) throws WorkflowException {
+      WorkflowModel workflowModel = null;
+      final String debugText = "Using configured delegate Workflow Model [ {} ] found in {}";
+      final String exceptionText = "Could not find configured Workflow Model [ %s ] in %s";
+
+      // Check /conf
+      if (StringUtils.isNotBlank(confWorkflowModelId)) {
+        workflowModel = getWorkflowModel(workflowSession, confWorkflowModelId);
+
+        if (workflowModel != null) {
+          log.debug(debugText, workflowModel.getId(), "/conf");
+        } else {
+          throw new WorkflowException(String.format(exceptionText, confWorkflowModelId, "/conf"));
         }
+        // Check Content
+      } else if (StringUtils.isNotBlank(contentWorkflowModelId)) {
 
-        log.debug("Provided PROCESS_ARGS: propertyName = [ {} ], Default Workflow Model = [ {} ]", propertyName, defaultWorkflowModelId);
-
-        ResourceResolver resolver = null;
-        WorkflowData wfData = workItem.getWorkflowData();
+        workflowModel = getWorkflowModel(workflowSession, contentWorkflowModelId);
 
-        if (!workflowHelper.isPathTypedPayload(wfData)) {
-            log.warn("Could not locate a JCR_PATH payload for this workflow. Skipping delegation.");
-            return;
+        if (workflowModel != null) {
+          log.debug(debugText, workflowModel.getId(), "/content");
+        } else {
+          throw new WorkflowException(String.format(exceptionText, confWorkflowModelId, "/content"));
         }
 
-        final String path = wfData.getPayload().toString();
+        // Check default
+      } else if (StringUtils.isNotBlank(defaultWorkflowModelId)) {
+        workflowModel = getWorkflowModel(workflowSession, defaultWorkflowModelId);
 
-        // Get the resource resolver
-
-        resolver = workflowHelper.getResourceResolver(workflowSession);
-        if (resolver == null) {
-            throw new WorkflowException("Could not adapt the WorkflowSession to a ResourceResolver. Something has gone terribly wrong!");
+        if (workflowModel != null) {
+          log.debug(debugText, workflowModel.getId(), "parameter defaultworkflowmodelid");
+        } else {
+          throw new WorkflowException(
+              String.format(exceptionText, confWorkflowModelId, "parameter defaultworkflowmodelid"));
```

Comment:
I dont think `defaultworkflowmodelid` this is the actual parameter name?  Better to use the Constant DEFAULT_WORKFLOW_MODEL.